### PR TITLE
Listeners for performed requests

### PR DESCRIPTION
**Pull request by giorgiosironi**
Currently we are able to logs requests and responses of failed interactions, but not to hook in to see which requests are being performed for testing or monitoring purposes. I wanted to do this in a way that is not too Guzzle dependent.

With this in place, we can specify exactly which requests should be triggered, whereas in https://github.com/elifesciences/api-sdk-php/pull/78 it seems that additional requests for pages not waited upon were invisible despite causing an exception.

### CODE REVIEW

**Comment by thewilkybarkid** [comment]
`NotifyingHttpClient`?

**Comment by giorgiosironi** [comment]
A decorator? Ok, will refactor towards it

File: test/HttpClient/Guzzle6HttpClientTest.php [hunk 1]
```diff
@@ -28,6 +28,7 @@ protected function setUp()
 
         $this->mock = new MockHandler();
         $this->guzzle = new Client(['handler' => HandlerStack::create($this->mock)]);
+        $this->client = new Guzzle6HttpClient($this->guzzle);
```
Comment (thewilkybarkid) [inline test/HttpClient/Guzzle6HttpClientTest.php]: Could you revert these changes? Don't disagree with them, but they're unrelated.
